backend/src/apiserver/server/util.go

```diff
@@ -76,18 +80,23 @@ func DecompressPipelineTarball(compressedFile []byte) ([]byte, error) {
 		return nil, util.NewInvalidInputErrorWithDetails(err, "Error extracting pipeline from the tarball file. Not a valid tarball file.")
 	}
 	tarReader := tar.NewReader(gzipReader)
-	header, err := tarReader.Next()
-	if err != nil || header == nil {
-		return nil, util.NewInvalidInputErrorWithDetails(err, "Error extracting pipeline from the tarball file. Not a valid tarball file.")
-	}
-	if !isYamlFile(header.Name) {
-		return nil, util.NewInvalidInputError("Error extracting pipeline from the tarball file. Expecting a YAML file inside the tarball. Got: %v", header.Name)
-	}
-	decompressedFile, err := ioutil.ReadAll(tarReader)
-	if err != nil {
-		return nil, util.NewInvalidInputErrorWithDetails(err, "Error reading pipeline YAML from the tarball file.")
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil || header == nil {
```

Comment:
No need to check header == nil. It will not be nil if err is non-nil

Reply:
You're right.
I wasn't changing this code, BTW - I just indented it.

Reply:
Still not changed?

Reply:
Done